swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -477,28 +488,61 @@ private ContractEntity transformEntitlementToContractEntity( // NOSONAR
 
   private void collectMissingUpStreamContractDetails( // NOSONAR
       ContractEntity entity, PartnerEntitlementContract contract) throws ApiException {
-    String awsCustomerAccountId = contract.getCloudIdentifiers().getAwsCustomerAccountId();
-    String productCode = contract.getCloudIdentifiers().getProductCode();
-    if (Objects.nonNull(contract.getCloudIdentifiers())
-        && Objects.nonNull(awsCustomerAccountId)
-        && Objects.nonNull(productCode)) {
-      PageRequest page = new PageRequest();
-      page.setSize(20);
-      page.setNumber(0);
-      log.trace(
-          "Call Partner Api to fill missing information using customerAwsAccountId {} and vendorProductCode {}",
-          awsCustomerAccountId,
-          productCode);
-      var result =
-          partnerApi.getPartnerEntitlements(
-              new QueryPartnerEntitlementV1()
-                  .customerAwsAccountId(awsCustomerAccountId)
-                  .vendorProductCode(productCode)
-                  .page(page));
-      mapUpstreamContractToContractEntity(entity, result);
+    String customerAccountId;
+    String productCode;
+    var marketplace = determineMarketplaceForContract(contract);
+
+    if (Objects.equals(marketplace, "aws")) {
```

Comment:
Does it make sense to use the ContractSourcePartnerEnum values here instead?

Reply:
Yes it does, Ill make the changes now